src/expressRequest.d.ts

```diff
@@ -0,0 +1,6 @@
+declare namespace Express {
+    export interface Request {
```

Comment:
This will make TS believe that every `req` object has an `id` and `name`, but this isn't true - these fields will get added only after passing through your validation middleware. Perhaps it would be better to create an extended Request like:
```ts
type ValidatedRequest = Request & {
    id: string,
    name: string
}
```
and use it in handler:
```ts
const validated = req as ValidatedRequest;
```
or use a type guard instead of `as`:
```ts
function isValidatedRequest(req: Request): req is ValidatedRequest {
  return req.id !== undefined && req.name !== undefined;
}
// ...
if (!isValidatedRequest(req)) {
  throw new Error();
}
```